internal/provider/newrelic/provider_integration_test.go

```diff
@@ -123,6 +125,58 @@ func Test_Getting_external_metric_generates_a_query_not_rejected_by_backend(t *t
 	})
 }
 
+//nolint:paralleltest // This test registers environment variables, so it must not be run in parallel.
+func Test_Getting_external_metric_through_proxy(t *testing.T) {
+	ctx := testutil.ContextWithDeadline(t)
+
+	port := 1337
+	t.Setenv("HTTPS_PROXY", fmt.Sprintf("localhost:%d", port))
+
+	t.Run("ends_with_error_when_proxy_fails", func(t *testing.T) {
+		proxy := runProxy(t, port, func(host string, ctx *goproxy.ProxyCtx) (*goproxy.ConnectAction, string) {
+			return goproxy.MitmConnect, host
+		})
+
+		t.Cleanup(func() {
```

Comment:
could cleanup be moved inside runProxy?

Reply:
Fixed in 34b65a2a2eaaa01c9382813cd910343c97f7c1dc